GHzDACs/GHzAD_v7.md

```diff
@@ -0,0 +1,437 @@
+    d(8)    sram(+6)[7..0]    spare
+    d(9)    sram(+7)[7..0]    spare
+
+    d(10)   sram(+8)[7..0]    rcount[7..0](1)
+    ...
+    d(1022)sram(+1023)[7..0]  rcount[15..8](127)
+    ...
+    d(1025)sram(+1023)[7..0]  spare
+
+#### Mixer table
+
+For the multiplier lookup tables, adrstart is taken from the following table
+adrstart[20..8]= channel n + 1.
+
+This offsets by 1 from the above trigger page.
+
+The Ethernet packet for channel n is (in signed int (‘<i1’ for python or ‘int8’):
+
+    l(0)    length[15..8]     set to 4; ( length[15..0]= 1024+2=1026 )
+    l(1)    length[7..0]      set to 2
+
+    d(0)    adrstart[15..8]   SRAM page for start address: middle bits = n+1 
+    d(1)    adrstart[23..16]  upper bits; size of SRAM implies bits [23..19] = 0
+
+    d(2)    sram(+0)[7..0]    multsin(0)	I0	Multiplier time 0
+    d(3)    sram(+1)[7..0]    multcos(0)	Q0
+    d(4)    sram(+2)[7..0]    multsin(1)	I1	Multiplier time 1 (1/2 clock, 2ns)
+    d(5)    sram(+3)[7..0]    multcos(1)	Q1
+    ...
+    d(1024)sram(+1022)[7..0]  multsin(511)
+    d(1025)sram(+1023)[7..0]  multcos(511)
+
+512 * ns = 1.024us max demod length
+
+
+
+## Register Write
+
+These registers control the AD functions.
+This card is always set in slave mode for daisychain initiation of AD functions (see GHzDAC board).  
+
+    l(0)    length[15..8]   set to 0  ; ( length[15..0] = 59 )
+    l(1)    length[7..0]    set to 59
+
+    d(0)    start[7..0]        Output & command function
+                               0 = off
+                               1 = register readback
+                               2 = average mode, auto start (use n=1)
+                               3 = average mode, daisychain start
+                               4 = demodulator mode, auto start (use n=1)
+                               5 = demodulator mode, daisychain start
+                               6 = set PLL of 1GHz clock with ser[23..0], no readback
+                               7 = recalibrate AD converters, no readback
+    d(1)    startdelay[7..0]   Start delay after daisychain signal, compensates for dchain delays.
+    d(2)    startdelay[15..8]      Note longer delays than for GHzDAC
+    d(3)    ser1[7..0]         8 lowest PLL bits of serial interface 
+    d(4)    ser2[7..0]         8 middle PLL bits
+    d(5)    ser3[7..0]         8 highest PLL bits
+    d(6)    spare
+
+    d(7)    n[7..0]	           n = Number of averages in average mode
+    d(8)    n[15..8]           n = Number of total events in demodulator mode
+    d(9)    bitflip[7..0]      XOR mask for bit readout	
+    d(10)   mon0[7..0]         SMA mon0 and mon1 programming, like DAC board
+    d(11)   mon1[7..0]
+    ...
+    d(58)   spare
+
+
+## Register Readback
+
+Registers are readback according to the above data fields and additional bytes given here.
+Length of readback = 46.
+
+    d(0)    build[7..0]         Build number of FPGA code.  
```

Comment:
All of the other packet formats (both for writes and readbacks) have the length bytes l(0) and l(1) listed here.  Either way is fine (I prefer without), but be consistent.